store/store.go

```diff
@@ -1119,6 +1128,31 @@ func findRev(needle snap.Revision, haystack []snap.Revision) bool {
 // The file is saved in temporary storage, and should be removed
 // after use to prevent the disk from running out of space.
 func (s *Store) Download(name string, downloadInfo *snap.DownloadInfo, pbar progress.Meter, user *auth.UserState) (path string, err error) {
+
+	if useDeltas() {
+		logger.Debugf("Available deltas returned by store: %v", downloadInfo.Deltas)
+	}
+	if useDeltas() && len(downloadInfo.Deltas) == 1 {
+		workingDir, err := ioutil.TempDir("", name+"-deltas")
+		deltaInfo := &downloadInfo.Deltas[0]
+		if err == nil {
+			defer os.RemoveAll(workingDir)
+
+			deltaPath, err := s.downloadDelta(name, workingDir, downloadInfo, pbar, user)
+			if err == nil {
+				logger.Debugf("Successfully downloaded delta for %s at %s", name, deltaPath)
+				snapPath := ""
+				snapPath, err = applyDelta(name, deltaPath, deltaInfo)
+				if err == nil {
+					logger.Debugf("Successfully applied delta for %s at %s. Returning %q instead of full download and saving %d bytes.", name, deltaPath, snapPath, downloadInfo.Size-deltaInfo.Size)
+					return snapPath, nil
```

Comment:
You missed my // TODO comment below :-). I'd like to do that in a separate branch - remember the generated snap is still verified against the assertions, so unless I'm missing something, worst case is that iff the env var is explicitly set and a bad delta download occurs, either the xdelta will error or if not, the assertions won't match. FWIW, I don't see where the sha3_384 is checked for the full download either (but perhaps that's handled by the assertion checking).